### PR TITLE
CGAL_Core: make it less verbose to fix #1915

### LINKED ISSUE #1915: Core is too verbose
There are some `std::cerr` in Core and also the file `Core_Diagnotics` that is created on the disk when an error occurs. I think they should be protected by the macro `CORE_TRACE`. Do you agree?
While at it, shall we rename the debug macro `CORE_TRACE` to `CGAL_CORE_TRACE`?

### PR DESCRIPTION
**Pull request by afabri**

### CODE REVIEW

File: CGAL_Core/include/CGAL/CORE/BigFloat_impl.h [hunk 1]
```diff
@@ -137,7 +137,7 @@ BigFloatRep::BigFloatRep(double d) : m(0), err(0), exp(0) {
       exp--;
       stop++;
     }
-#ifdef CORE_DEBUG
+#ifdef CGAL_CORE_DEBUG
     CGAL_assertion (s >= 0);
 #endif
```
Comment (lrineau) [inline CGAL_Core/include/CGAL/CORE/BigFloat_impl.h]: (That is just a remark, and not a request for a change.)

I find strange that such an assertion is protected by macro `CORE_DEBUG`, knowing that `CGAL_assertion` itself it protected by `NDEBUG` and `CGAL_NDEBUG`.

**Comment by afabri** [comment]
@lrineau Any idea why this fails?  It does not do on VC++ on my machine.

**Comment by lrineau** [comment]
@afabri, in https://github.com/CGAL/cgal/pull/2017#issuecomment-291422651:
> @lrineau Any idea why this fails?  It does not do on VC++ on my machine.

What do you mean by it fails? If that is shown by the testsuite, then please copy-paste a link to the error log.

**Comment by afabri** [comment]
When I had written that it failed travis was red. Now it is green.  A miracle.

**Comment by lrineau** [comment]
Sometime Travis jobs fails for networks errors, and it happens that I restart jobs that failed that way.

**Pull request merged**